pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -88,6 +93,33 @@ func (ku *Upgrader) RunUpgrade() error {
 	return ku.upgradeAgentPools(ctx)
 }
 
+// handleUnreconcilableAddons ensures addon upgrades that addon-manager cannot handle by itself.
+// This method fails silently otherwide it would break test "Should not fail if a Kubernetes client cannot be created" (upgradecluster_test.go)
+func (ku *Upgrader) handleUnreconcilableAddons() {
+	// kube-proxy upgrade fails from v1.15 to 1.16: https://github.com/Azure/aks-engine/issues/3557
+	// deleting daemonset so addon-manager recreates instead of patching
+	upgradeVersion := ku.DataModel.Properties.OrchestratorProfile.OrchestratorVersion
+	if !common.IsKubernetesVersionGe(ku.CurrentVersion, "1.16.0") && common.IsKubernetesVersionGe(upgradeVersion, "1.16.0") {
+		ku.logger.Infof("Attempting to delete kube-proxy daemonset.")
+		client, err := ku.getKubernetesClient(getResourceTimeout)
+		if err != nil {
+			ku.logger.Errorf("Error getting Kubernetes client: %v", err)
+			return
+		}
+		ds, err := client.GetDaemonSet("kube-system", common.KubeProxyAddonName)
```

Comment:
Do we need to bother with `GetDaemonSet`, or can we just call `DeleteDaemonSet` and handle the possible error if it didn't exist?

Reply:
can update